src/log/log.c

```diff
@@ -2387,6 +2400,15 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
             if (LF_ISSET(WT_LOGSCAN_ONE))
                 break;
         }
+
+        /*
+         * Exit the scanning loop if the next LSN seen is greater than our user set end range LSN.
+         */
+        if (end_range_lsnp != NULL &&
+          ((next_lsn.l.file > end_range_lsnp->l.file) ||
+            (next_lsn.l.file == end_range_lsnp->l.file &&
+              next_lsn.l.offset > end_range_lsnp->l.offset)))
```

Comment:
You should use `__wt_log_cmp` to compare LSNs.

Reply:
Great, yeah that's much neater